src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -140,7 +141,7 @@ public function dump(array $options = array())
 
         $code = $this->startClass($options['class'], $options['base_class'], $options['namespace']);
 
-        if ($this->container->isFrozen()) {
+        if ($this->container->getParameterBag() instanceof FrozenParameterBag) {
```

Comment:
Havent figured it out yet, i think so.

[Tests](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/Tests/Dumper/PhpDumperTest.php#L123) somewhat imply this happens for compiled container builders.